tmt/steps/discover/fmf.py

```diff
@@ -301,6 +301,26 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
             if not url:
                 assert_git_url(self.step.plan.name)
 
+        # for imported plans print details about a remote plan
+        remote_plan = getattr(
+            getattr(
+                getattr(
+                    self.parent,
+                    'plan',
+                    None),
+                '_original_plan',
+                None),
+            '_remote_plan_fmf_id',
+            None)
```

Comment:
I don't understand the need for so many `getattr()` calls. Or maybe I do, but I wouldn't choose this road, as it makes it harder for linters and other tools to follow the tree of classes and annotations. `remote_plan` here is simply `Any` or `None`, that's the best mypy can tell us. Should we make a typo & ask for `_original_pln`, nothing would stop us. It works, it's definitely possible, but my two cents, it's not suitable for a project as complicated as tmt anymore. My proposal would be something along these lines, maybe a `cast()` would be needed, but it still should be more readable for humans and linters:

```python
parent = cast(Optional[Discover], self.parent)
if parent and parent.plan._original_plan:
    remote_plan_id = self.parent.plan._original_plan._remote_plan_fmf_id

    for key, value in remote_plan_id.to_minimal_spec().items():
        # Already printed out as "ref"
        if key == '_ref_expanded':
          continue

        if key == 'ref':
            # There should be no need for getattr(), _ref_expanded is fully declared and assigned
            # a default value - it *must* be present.
            value = remote_plan_id._ref_expanded

        # Is replace() needed? No key name contains an underscore, _ref_expanded is apparently skipped.
        # And no need for `if value is not None` checks as keys with such values are not presented by to_minimal_spec()
        self.info(f'imported-{key}', value, 'green')
```

Reply:
I know `getattr` is ugly, I just don't know which of those attributes exist. Now looking at `class Plan` it seems it should be fine to follow the approach you have above. Thank you.

Reply:
I have updated the code following your approach. Thank you.